### PR TITLE
Test with fresh venv

### PR DESCRIPTION
**Pull request by fayvor**
[#156](https://github.com/ibm-granite-community/pm/issues/156)
- Test each notebook in a separate virtualenv; requires per-notebook ipykernel install
- Update pre-commit hook to preserve cell 'scrolled' flag and remove python version info

### PR Checklist

- [ ] **Commits signed**: All commits must be GPG or SSH signed.
- [ ] **DCO Compliance**: Developer Certificate of Origin (DCO) applies to the code, documentation, and any example data provided. Ensure commits are signed off.
- [ ] **Notebook outputs cleared**: Ensure all notebook outputs are cleared.
- [ ] **Automated testing**: Add the recipe to the automated tests.
- [ ] **Test in Google Colab**:
    - [ ] Test that it works in Google Colab (Python 3.10.12).
    - [ ] Colab has its own package set and Python version, so ensure compatibility.
- [ ] **Test locally**:
    - [ ] Ensure the code works in a fresh Python virtual environment (venv).
- [ ] **Standard access to secrets and variables** Include `!pip install git+https://github.com/ibm-granite-community/utils` in the first code cell in order to make `get_env_var` available to accessing secrets and variables in the recipe.
- [ ] **Flexible LLM platform support** The platform should be easily switchable. Use LangChain or LlamaIndex.
- [ ] **Example data**: Follow the example data guidance.
- [ ] **README.md updates**:
    - [ ] Add a link to the recipe in the Table of Contents (ToC).
    - [ ] Include a Colab button after that link.

### CODE REVIEW

**Comment by deanwampler** [comment]
Reviewing...

**Review by deanwampler — CHANGES_REQUESTED** [review]
I'm running the script in the `workflows.yaml` locally. It worked for a while, except `recipes/Time_Series/Bike_Sharing_Finetuning_with_Exogenous.ipynb` ran for at least 30 minutes, then failed with "too many open files". Also, the script kept running to the next notebook in the loop. The error didn't terminate the whole test run.

Maybe the notebook failure is some oddity of my local environment, but it looks like _all_ our test scripts in `*_workflows.yaml` should start with `set -e`, so failures cause the whole thing to fail.

**Comment by deanwampler** [comment]
One other suggestion, the `snack box` equivalent `vanilla_workflows.yaml`, uses this to read the list of notebooks:
```
notebooks=$(cat .github/notebook_lists/vanilla_notebooks.txt | grep -v '^#' | tr '\n' ' ')
```

The extra filtering for `^#` lets people comment out a notebook and it will be skipped.

File: .github/workflows/vanilla_workflows.yaml [hunk 1]
```diff
@@ -46,7 +46,16 @@ jobs:
           REPLICATE_API_TOKEN: ${{ secrets.REPLICATE_API_TOKEN }}
         run: |
           notebooks=$(cat .github/notebook_lists/vanilla_notebooks.txt | tr '\n' ' ')  
-          for notebook in $notebooks; do  
-            echo "Executing $notebook"  
-            jupyter nbconvert --to notebook --execute --inplace $notebook  
+          for notebook in $notebooks; do 
+            echo "Building virtual env"
```
Comment (rawkintrevo) [inline .github/workflows/vanilla_workflows.yaml:50]: Two questions:
1) Why are you making a virtual environment?
2) Why are you doing it here, not in it's own step?

**Pull request merged**